http-client/test/Network/HTTP/ClientSpec.hs

```diff
@@ -46,6 +49,33 @@ spec = describe "Client" $ do
             res <- httpLbs req man
             responseStatus res `shouldBe` found302
 
+    context "managerModifyResponse" $ do
+      it "allows to modify the response status code" $ do
+        let modify :: Response BodyReader -> IO (Response BodyReader)
+            modify res = do
+              return res {
+                responseStatus = (responseStatus res) {
+                  statusCode = 201
+                }
+              }
+            settings = defaultManagerSettings { managerModifyResponse = modify }
+        man <- newManager settings
+        res <- httpLbs "http://httpbin.org" man
+        (statusCode.responseStatus) res `shouldBe` 201
+
+      it "modifies the response body" $ do
+        let modify :: Response BodyReader -> IO (Response BodyReader)
+            modify res = do
+              return res {
+                responseBody = do
+                  reader <- constBodyReader [BS.pack "modified response body"]
```

Comment:
Oh. Makes sense, now that you pointed me at it.

Out of curiosity. Who is continuously accessing `responseBody` in the test case?